### PR TITLE
Prepare for release v1.16.0-pre.3

### PR DESCRIPTION
**Pull request by aanm**
Prepare for release v1.16.0-pre.3

See the included CHANGELOG.md for a full list of changes.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -28,6 +29,7 @@ Alexey Grevtsev                         alexey.grevtcev@gmail.com
 Alex Katsman                            alexkats@google.com
 Alex Romanov                            alex@romanov.ws
 Alex Szakaly                            alex.szakaly@gmail.com
+alex waring                             ajmwaring@gmail.com
 Alex Waring                             alex.waring@starlingbank.com
```
Comment (christarazi) [inline AUTHORS]: Dup? Can be cleaned up later

File: AUTHORS [hunk 2]
```diff
@@ -285,6 +289,7 @@ Glen Yu                                 glen.yu@gmail.com
 Glib Smaga                              code@gsmaga.com
 Gobinath Krishnamoorthy                 gobinathk@google.com
 Gowtham Sundara                         gowtham.sundara@rapyuta-robotics.com
+gray                                    greyschwinger@gmail.com
 Gray Lian                               gray.liang@isovalent.com
```
Comment (christarazi) [inline AUTHORS]: Same here

**Review by christarazi — APPROVED** [review]

File: AUTHORS (hunk 1 above)
Comment (joestringer) [reply AUTHORS]: We can fix this in `.mailmap`.

File: Documentation/network/kubernetes/compatibility-table.rst [hunk 3]
```diff
@@ -68,13 +118,9 @@
...
 +--------------------+----------------+
-| v1.15              | 1.28.2         |
+| v1.15.5            | 1.28.2         |
 +--------------------+----------------+
-| v1.16.0-pre.0      | 1.29.2         |
-+--------------------+----------------+
-| v1.16.0-pre.1      | 1.29.4         |
-+--------------------+----------------+
-| v1.16.0-pre.2      | 1.29.5         |
+| v1.15              | 1.28.2         |
```
Comment (joestringer) [inline Documentation/network/kubernetes/compatibility-table.rst:123]: This looks like `check-crd-compat-table.sh` was run with v1.15 as the argument, so it incorrectly included v1.13 versions and excluded v1.16 (which is this target version).

I've fixed it by running the `main` version of the script from the v1.16 preparation dir with `check-crd-compat-table.sh v1.16 --update`.

Comment (joestringer) [reply Documentation/network/kubernetes/compatibility-table.rst:123]: (No I didn't fix it, this was merged before I could fix it)

Comment (joestringer) [reply Documentation/network/kubernetes/compatibility-table.rst:123]: Updated here: https://github.com/cilium/cilium/pull/32859

**Pull request merged**